packages/langium/src/generator/generator-node.ts

```diff
@@ -17,6 +17,13 @@ export const EOL = (typeof process === 'undefined') ? '\n' : (process.platform =
 export type Generated = GeneratorNode | string | undefined;
 export type GeneratorNode = CompositeGeneratorNode | IndentNode | NewLineNode;
 
+export interface IndentConfig {
+    indentation?: string|number;
+    indentedChildren?: Generated[] | ((indented: IndentNode) => void);
```

Comment:
Now that this is wrapped in an object, you could as well separate these variants in two properties `indentedChildren` and `addChildren` or similar.

Reply:
Right. On the other hand I like the similarity to the non-object-based definition, but that's maybe personal taste.

I'm just adding some missing documentation to `joinToNode` and a few additional test cases, than it's done I hope...